tests/trainer/test_dataloaders.py

```diff
@@ -396,24 +378,22 @@ def test_datasets_dataloaders_with_limit_num_batches(
 
     trainer.fit(model, train_dataloaders=train_dl, val_dataloaders=val_dl)
     assert trainer.state.finished, f"Training failed with {trainer.state}"
-    assert trainer.num_training_batches == (limit_train_batches if limit_train_batches > 0.0 else float("inf"))
-    if limit_val_batches != 0.0:
-        assert trainer.num_val_batches[0] == limit_val_batches
-    else:
-        assert trainer.num_val_batches == []
-    assert epoch_cb.train_epoch_count == (epochs if limit_train_batches > 0 else 0)
-    assert epoch_cb.train_batches_seen == limit_train_batches * epochs
-    assert epoch_cb.val_epoch_count == (epochs if limit_val_batches > 0 else 0)
-    assert epoch_cb.val_batches_seen == limit_val_batches * epochs
+    assert trainer.num_training_batches == limit_batches
+    assert trainer.num_val_batches[0] == limit_batches
+    assert epoch_cb.train_epoch_count == max_epochs
+    assert epoch_cb.train_batches_seen == limit_batches * max_epochs
+    assert epoch_cb.val_epoch_count == max_epochs
+    assert epoch_cb.val_batches_seen == limit_batches * max_epochs
 
     trainer.test(model, dataloaders=test_dl)
-    assert trainer.num_test_batches[0] == limit_test_batches
-    assert epoch_cb.test_epoch_count == int(limit_test_batches > 0)
+    assert trainer.num_test_batches[0] == limit_batches
+    assert epoch_cb.test_epoch_count == 1
 
 
+@pytest.mark.skip()
 @pytest.mark.parametrize(
     ["limit_train_batches", "limit_val_batches", "limit_test_batches"],
-    [(0.0, 0.0, 0.0), (0, 0, 0.5), (1.0, 1.0, 1.0), (0.2, 0.4, 0.4)],
```

Comment:
should we group the new test together with the tests in this file? It might not be obvious that the removed cases are covered by a different test function.